File: java/src/ResponderBuilder.java


class ResponderBuilder {
	public static ResultResponder responder_for(Request request) {
		if (request.request_type() == "GET")
			return new WikiPageResponder();
		if (request.request_type() == "POST") {
			if (request.data().containsKey("search_text")) {
				if (request.data().containsKey("replace")) {
					return new SearchReplaceResponder();
				} else {
					return new SearchResponder();
				}
			}
			if (request.data().containsKey("where_used")) {
				return new WhereUsedResponder();
			}
			if (request.data().containsKey("tags")) {
				return new PropertySearchResponder();
			}
		}
		throw new IllegalArgumentException();
	}
}

File: csharp/ResponderBuilder.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class ResponderBuilder
	{
		public static ResultResponder ResponderFor(Request request)
		{
			if (request.RequestType == "GET")
				return new WikiPageResponder();
			if (request.RequestType == "POST")
			{
				if (request.Data.ContainsKey("search_text"))
				{
					if (request.Data.ContainsKey("replace"))
						return new SearchReplaceResponder();
					else
						return new SearchResponder();
				}
				if (request.Data.ContainsKey("where_used"))
					return new WhereUsedResponder();
				if (request.Data.ContainsKey("tags"))
					return new PropertySearchResponder();
			}
			throw new InvalidOperationException();
		}
	}
}

